src/Centreon/Infrastructure/HostConfiguration/Repository/HostConfigurationRepositoryRDB.php

```diff
@@ -866,4 +868,30 @@ private function updateMonitoringServerRelation(int $hostId, int $monitoringServ
         $statement->bindValue(':host_id', $hostId, \PDO::PARAM_INT);
         $statement->execute();
     }
+
+    /**
+     * Add or update the link between a host and a host severity
+     *
+     * @param Host $host
+     */
+    private function updateHostSeverity(Host $host): void
+    {
+        if (empty($host->getSeverities())) {
+            return;
+        }
+
+        $request = $this->translateDbName(
+            "DELETE `:db`.hostcategories_relation
+            FROM `:db`.hostcategories_relation
+            JOIN `:db`.hostcategories
+            ON hostcategories.hc_id = hostcategories_relation.hostcategories_hc_id
+            WHERE hostcategories.level IS NOT NULL
+            AND hostcategories_relation.host_host_id = :host_id"
+        );
+        $statement = $this->db->prepare($request);
+        $statement->bindValue(':host_id', $host->getId(), \PDO::PARAM_INT);
+        $statement->execute();
+
```

Comment:
return must be after deletion
otherwise, you cannot create a hos without severity
```suggestion
        $request = $this->translateDbName(
            "DELETE `:db`.hostcategories_relation
            FROM `:db`.hostcategories_relation
            JOIN `:db`.hostcategories
            ON hostcategories.hc_id = hostcategories_relation.hostcategories_hc_id
            WHERE hostcategories.level IS NOT NULL
            AND hostcategories_relation.host_host_id = :host_id"
        );
        $statement = $this->db->prepare($request);
        $statement->bindValue(':host_id', $host->getId(), \PDO::PARAM_INT);
        $statement->execute();

        if (empty($host->getSeverities())) {
            return;
        }
```

Reply:
This is an update not a creation, host already exists and possibly have a severity.
If no valid severity is found from the discovery mappers the existing one should not be deleted.